charts/kong/README.md

```diff
@@ -146,13 +146,22 @@ read the [env](#the-env-section) section.
 
 When deploying Kong in DB-less mode(`env.database: "off"`)
 and without the Ingress Controller(`ingressController.enabled: false`),
-you have to provide a declarative configuration for Kong to run.
-The configuration can be provided using an existing ConfigMap
-(`dblessConfig.configMap`) or or the whole configuration can be put into the
-`values.yaml` file for deployment itself, under the `dblessConfig.config`
-parameter. See the example configuration in the default values.yaml
-for more details.
-
+you have to provide a declarative configuration for Kong to run. [Declarative
+configuration](https://docs.konghq.com/gateway-oss/2.5.x/db-less-and-declarative-config/#the-declarative-configuration-format)
+is supplied via ConfigMap.    
+    
+Possible sources of the declarative configuration ConfigMap include configuration 
+supplied under the `dblessConfig.config` paramater of `values.yaml`, or with the
+helm flag `--set-file` parameter. See the example configuration in the default 
+[values.yaml](values.yaml#L371) or [example-values/dbless-config.yaml](example-values/dbless-config.yaml)
+for syntax and more details.    
+    
+Alternatively, the configuration can be provided separately using an existing
+ConfigMap name supplied to Kong in (`dblessConfig.configMap`). Note, externally
+supplied ConfigMaps are not hashed or tracked in deployment annotations.
+Subsequent ConfigMap updates will require user-initiated new deployment rollouts
+to apply the new configuration.    
```

Comment:
- Latest doc since we don't usually scan through docs to bump versions.
- values.yaml locations change all the time, so only including the key names here.
- The example-values directory is for values.yamls only rather than general example content. I recommend omitting any separate example here since the linked doc page already includes its own example, which doesn't need to change at all for `--set-file` use.
- Reworded the ConfigMap rollout warning to apply across the board. The template version/`dblessConfig.config` only updates the generated ConfigMap and won't change the Kong Deployment, so Helm won't perform its own rollout if that's the only thing you're changing.
 
```suggestion
you have to provide a [declarative configuration]https://docs.konghq.com/gateway-oss/latest/db-less-and-declarative-config/#the-declarative-configuration-format) for Kong to run.
You can provide an existing ConfigMap
(`dblessConfig.configMap`) or place the whole configuration into
`values.yaml` (`dblessConfig.config`)
parameter. See the example configuration in the default values.yaml
for more details. You can use `--set-file dblessConfig.config=/path/to/declarative-config.yaml`
in Helm commands to substitute in a complete declarative config file.
    
Note that ConfigMap updates via either method will not restart Kong pods,
and they will not recognize updated configuration until restarted. You should run
`kubectl rollout restart deploy` after updating the configuration ConfigMap contents
or changing `dblessConfig.config`.
```

Reply:
I believe 5848f6e & 27e8aee resolves these items.